src/src/com/microsoft/aad/adal/StorageHelper.java

```diff
@@ -632,23 +618,30 @@ private SecretKey unwrap(Cipher wrapCipher, byte[] keyBlob) throws GeneralSecuri
         }
     }
 
-    private static void writeKeyData(File file, byte[] data) throws IOException {
+    private void writeKeyData(final byte[] data) throws IOException {
         Logger.v(TAG, "Writing key data to a file");
-        final OutputStream out = new FileOutputStream(file);
+        final File keyFile = new File(mContext.getDir(mContext.getPackageName(), Context.MODE_PRIVATE),
+                ADALKS);
+        final OutputStream out = new FileOutputStream(keyFile);
         try {
             out.write(data);
         } finally {
             out.close();
         }
     }
 
-    private static byte[] readKeyData(File file) throws IOException {
+    private byte[] readKeyData() throws IOException {
+        final File keyFile = new File(mContext.getDir(mContext.getPackageName(), Context.MODE_PRIVATE),
+                ADALKS);
+        if (!keyFile.exists()) {
+            throw new IOException("Key file to read does not exist");
+        }
+        
         Logger.v(TAG, "Reading key data from a file");
-        final InputStream in = new FileInputStream(file);
-
+        final InputStream in = new FileInputStream(keyFile);
         try {
-            ByteArrayOutputStream bytes = new ByteArrayOutputStream();
-            byte[] buffer = new byte[1024];
+            final ByteArrayOutputStream bytes = new ByteArrayOutputStream();
+            final byte[] buffer = new byte[1024];
```

Comment:
Well, it's assuming the byte array size is never more than 1kb, This is a reasonable assume, the wrapped key we store in the file shouldn't be over 1kb, and it saves us from reading the file size.